test/test/alloc/AllocTests.java

```diff
@@ -55,4 +59,16 @@ public void humongous(TestProcess p) throws Exception {
         assert out.contains("java/io/ByteArrayOutputStream.toByteArray;");
         assert out.contains("G1CollectedHeap::humongous_obj_allocate");
     }
+
+    @Test(mainClass = MapReaderOpt.class, jvmVer = {11, Integer.MAX_VALUE})
+    public void objectSamplerWtihDifferentAsprofs(TestProcess p) throws Exception {
+        Thread.sleep(1000);
+        Output out = p.profile("-e alloc -d 3 -o collapsed");
+        // _[k] suffix in collapsed output corresponds to jdk.ObjectAllocationOutsideTLAB, which means alloc tracer is being used
+        assert !out.contains("_\\[k\\]"); // we are using alloc tracer instead of object sampler, should definitely not happen on first profiling call
+        Path asprofCopy = Path.of("/tmp/libasyncProfiler." + p.currentOs().getLibExt());
```

Comment:
Temporary file should have a unique name (not to clash with an existing library and to be able to run multiple test processes simultaneously). The file should be deleted on exit.